packages/subgraph/src/liquidity-rewards/updaters.ts

```diff
@@ -275,61 +275,56 @@ export function linkBidToRewards(bid:Bid) : void {
   for(let i = 0; i < activeRewardIds.length; i++){
 
     let allocationRewardId = activeRewardIds[i];
-
  
     let rewardAllocation = RewardAllocation.load(allocationRewardId)!;
- 
 
     if( 
+        allocationStatusToEnum(rewardAllocation.status) == AllocationStatus.Active &&
+        bidIsEligibleForReward(bid,rewardAllocation)   
+    ){ 
+      appendAllocationRewardToBidParticipants(bid,rewardAllocation);       
+    }
 
-        bidIsEligibleForReward(bid,rewardAllocation)
-   
-    ){
- 
-      appendAllocationRewardToBidParticipants(bid,rewardAllocation);
+  }
 
-       
-    }
+}
 
 
+export function unlinkBidsFromReward(reward:RewardAllocation) : void {
 
+  let bidRewards = reward.bidRewards ; 
 
+  for(let i = 0; i < bidRewards.length; i++){
```

Comment:
unfortunately this isn't going to work because `reward.bidRewards` is a derived array 😕